packages/notebook/src/actions.tsx

```diff
@@ -372,7 +376,14 @@ export namespace NotebookActions {
 
     const newIndex = notebook.activeCell ? notebook.activeCellIndex : 0;
     model.sharedModel.insertCell(newIndex, {
-      cell_type: notebook.notebookConfig.defaultCell
+      cell_type: notebook.notebookConfig.defaultCell,
+      metadata:
+        notebook.notebookConfig.defaultCell === 'code'
+          ? {
+              // This is an empty cell created by user, thus is trusted
+              trusted: true
+            }
+          : {}
```

Comment:
This ended up a bit repetitive. A tempting alternative is to modify `sharedModel.insertCell` to always set `trusted: true` in metadata if cell type is `code` and `outputs` are empty. A potential downside of that alternative is that it is not "secure by default", because an innocent extension author might do:

```ts
model.sharedModel.insertCell(0, {cell_type: 'code'});
// and some lines later in the same block of instructions
notebook.activeCellIndex = 0;
notebook.activeCell.model.outputs = [anOutputLackingProperSanitisation];
```

in which case a cell with output potentially not controlled by user would end up marked as trusted.